### PR TITLE
[xray-exporter] Move JIRA fields mapping into JIRA configuration

### PR DESCRIPTION
**Pull request by uarlouski**
Closes https://github.com/vividus-framework/vividus/issues/2390

### CODE REVIEW

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java [hunk 1]
```diff
@@ -116,45 +103,8 @@ public JiraClient getByIssueKey(String issueKey) throws JiraConfigurationExcepti
      */
     public JiraClient getByProjectKey(String projectKey) throws JiraConfigurationException
     {
-        isTrue(projectKey != null, "The JIRA project key is not specified");
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java:119]: I don't think it should be removed

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraConfigurationProvider.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                return jiraConfiguration.get();
+            }
+            throw error("Unable to find JIRA configuration with '%s' key", key);
+        }
+
+        return getDefaultOrElseThrow(() -> error("The JIRA configuration key is not specified"));
+    }
+
+    JiraConfiguration getConfigurationByProjectKey(String projectKey) throws JiraConfigurationException
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/JiraConfigurationProvider.java:39]:
```suggestion
    JiraConfiguration getByProjectKey(String projectKey) throws JiraConfigurationException
```

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.vividus.jira;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.vividus.jira.model.JiraConfiguration;
+
+public class JiraFieldsMappingProvider
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java:8]: is it really needed? what's the profit to keep `JiraConfigurationProvider` package-private?

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraConfigurationProvider.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
+package org.vividus.jira;
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/JiraConfigurationProvider.java:1]:
```suggestion
package org.vividus.jira.config;
```
? and move to this package everything related to the configuration?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java (hunk 3 above)
Comment (uarlouski) [reply vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java:8]: JiraConfiguration is a mutable internal thing that contains configs that's why I think it should not be exposed as API provided to end-users and this is the reason why I created safe JiraFieldsMappingProvider

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java (hunk 1 above)
Comment (uarlouski) [reply vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java:119]: moved into JiraConfigurationProvider

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java (hunk 3 above)
Comment (valfirst) [reply vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java:8]: JiraConfigurationProvider.getFieldsMappingByProjectKey

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java [hunk 5]
```diff
@@ -104,7 +89,9 @@ public JiraClient getByJiraConfigurationKey(Optional<String> jiraConfigurationKe
     public JiraClient getByIssueKey(String issueKey) throws JiraConfigurationException
     {
         isTrue(issueKey != null, "The JIRA issue key is not specified");
-        return getClientByProjectKey(StringUtils.substringBefore(issueKey, "-"));
+        JiraConfiguration configuration = jiraConfigurationProvider
+                .getConfigurationByProjectKey(StringUtils.substringBefore(issueKey, "-"));
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java:93]: JiraConfigurationProvider.getByIssueKey

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java (hunk 3 above)
Comment (uarlouski) [reply vividus-facade-jira/src/main/java/org/vividus/jira/JiraFieldsMappingProvider.java:8]: hmm okay, this is better as long as we have only one method to retrieve fields mapping and not much of business logic

File: docs/modules/integrations/partials/jira-configuration.adoc [hunk 6]
```diff
@@ -6,23 +6,38 @@
...
+|The regular expression to match JIRA project keys.
+
+|*endpoint*
+|The JIRA Endpoint.
+
+|any of {allowed-http-properties}
+|The HTTP properties.
+
+|fields-mapping
+|The mapping between human-readable fields used by Vividus and fields used by JIRA.
```
Comment (valfirst) [inline docs/modules/integrations/partials/jira-configuration.adoc:28]:
```suggestion
|The mapping between human-readable fields used by VIVIDUS and fields used by JIRA.
```

File: vividus-facade-jira/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 7]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (valfirst) [inline vividus-facade-jira/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: where is it used?

Comment (uarlouski) [reply vividus-facade-jira/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: to mock JiraConfigurationProvider in JiraClientProvider

**Review by valfirst — APPROVED** [review]

**Pull request merged**